pkg/storage/cache.go

```diff
@@ -32,19 +33,38 @@ func CreateCacheDatabaseDriver(storageConfig config.StorageConfig, log zlog.Logg
 			return nil, nil //nolint:nilnil
 		}
 
-		if name != constants.DynamoDBDriverName {
+		if name != constants.DynamoDBDriverName &&
+			name != constants.RedisDriverName {
 			log.Warn().Str("driver", name).Msg("remote cache driver unsupported!")
 
 			return nil, nil //nolint:nilnil
 		}
 
-		// dynamodb
-		dynamoParams := cache.DynamoDBDriverParameters{}
-		dynamoParams.Endpoint, _ = storageConfig.CacheDriver["endpoint"].(string)
-		dynamoParams.Region, _ = storageConfig.CacheDriver["region"].(string)
-		dynamoParams.TableName, _ = storageConfig.CacheDriver["cachetablename"].(string)
+		if name == constants.DynamoDBDriverName {
+			// dynamodb
+			dynamoParams := cache.DynamoDBDriverParameters{}
+			dynamoParams.Endpoint, _ = storageConfig.CacheDriver["endpoint"].(string)
+			dynamoParams.Region, _ = storageConfig.CacheDriver["region"].(string)
+			dynamoParams.TableName, _ = storageConfig.CacheDriver["cachetablename"].(string)
 
-		return Create("dynamodb", dynamoParams, log)
+			return Create(name, dynamoParams, log)
+		}
+
+		if name == constants.RedisDriverName {
```

Comment:
As a closure maybe?

func redisParams(storageConfig.Cachedriver) redisParams {
blah...
}

Create(name, redisParams(), log)

Reply:
Why? THB I don't see the point of having the Create() even in current implementation. Why not just call cache.NewRedisCache(redisParams, log) here?